invenio_circulation/api.py

```diff
@@ -132,6 +132,28 @@ def is_item_available_for_checkout(item_pid):
         return search_result.hits.total == 0
 
 
+def is_item_at_desk_available_for_checkout(item_pid, patron_pid):
+    """Return True if the ITEM_AT_DESK is available for loan, False instead."""
+    # Try first to accept the availability for given patron
+    item_exception_loans = 0
+    search = search_by_patron_item_or_document(
+        item_pid=item_pid,
+        patron_pid=patron_pid,
+        filter_states=["ITEM_AT_DESK"],
+    )
+    search_result = search.execute()
+    if ES_VERSION[0] >= 7:
+        item_exception_loans = search_result.hits.total.value
+    else:
+        item_exception_loans = search_result.hits.total
+
+    if item_exception_loans == 1:
+        return True
+
+    # Then use normal behaviour if it failed with patron
+    return is_item_available_for_checkout(item_pid)
```

Comment:
Just to confirm, is it correct that if the item for the patron is not `ITEM_AT_DESK` or there are multiple loans, basically the search returns `0` or more than `1`, you fallback to that?
Isn't it better to raise because maybe the system data are not correct?

Reply:
@ntarocco the search should not return more than 1. If it is the case, you have some problems with data (duplicates for example). And so you need to search the problem elsewhere.

What we do is: 

* in case of `ITEM_AT_DESK` loan, we permit the librarian to make the checkout on the item. That's why we do the search on the patron_pid.
* in case this didn't pass, we use the normal behaviour, and so we call `is_item_available_for_checkout`

This will prevent - indeed - any problem about checkout (because we permit the normal behaviour to happen).

So on our side, I think the logic is OK. And any other error should come from either configuration or system corruption.

What do you think?

Reply:
I see, yes you are right, I just checked the implementation of `is_item_available_for_checkout` and it is good! 👍